### PR TITLE
Compact Filters: test catching when behind Block Headers

### PR DESCRIPTION
**Pull request by UnknownCypher**
Make sure the manager attempts to catch up to the block headers when behind.

---
Help fund my contributions to this project at https://funding.server.ky/project/nakamoto

### CODE REVIEW

File: p2p/src/protocol/cbfmgr.rs [hunk 1]
```diff
@@ -1456,9 +1456,65 @@ mod tests {
...
+            matches!(
+                m,
+                NetworkMessage::GetCFHeaders(GetCFHeaders {
+                    start_height,
+                    stop_hash,
+                    ..
+                }) if (*start_height as usize) == (cfheader_height + 1) && stop_hash == &chain.last().block_hash()
+            )
+        });
+        assert_matches!(msg, Some(..), "expect GetCFHeaders request");
```
Comment (cloudhead) [inline p2p/src/protocol/cbfmgr.rs:1517]: It's simpler to call `expect()` on the result of `find()`

File: p2p/src/protocol/cbfmgr.rs [hunk 2]
```diff
@@ -1456,9 +1456,65 @@ mod tests {
...
+            Link::Outbound,
+            &time,
+            &tree,
+        );
+
+        let mut events = util::events(cbfmgr.upstream.drain());
+        let ev = events.find(|e| {
+            matches!(e, Event::Syncing{start_height, stop_hash, ..}
+                               if (*start_height as usize) == (cfheader_height + 1)
+                               && stop_hash == &chain.last().block_hash())
```
Comment (cloudhead) [inline p2p/src/protocol/cbfmgr.rs:1502]: move to a `tip` variable to be re-used below

Comment (cloudhead) [reply p2p/src/protocol/cbfmgr.rs:1502]: I'd also format this `matches!` like the one below, unfortunately, `rustfmt` doesn't format inside macro calls..

File: p2p/src/protocol/cbfmgr.rs [hunk 3]
```diff
@@ -1456,9 +1456,65 @@ mod tests {
...
+            header_height,
+            REQUIRED_SERVICES,
+            Link::Outbound,
+            &time,
+            &tree,
+        );
+
+        let mut events = util::events(cbfmgr.upstream.drain());
+        let ev = events.find(|e| {
+            matches!(e, Event::Syncing{start_height, stop_hash, ..}
```
Comment (cloudhead) [inline p2p/src/protocol/cbfmgr.rs:1500]:
```suggestion
            matches!(e, Event::Syncing { start_height, stop_hash, .. }
```

**Review by vincenzopalazzo — APPROVED** [review]
LGTM

ack https://github.com/cloudhead/nakamoto/pull/73/commits/7ead737fd352c00a2cafe176d711f7051f2a5897

**Review by cloudhead — APPROVED** [review]

**Comment by cloudhead** [comment]
Merged in c005e022614d2ac33150b2711ffb49843c0562c9

**Pull request closed without merging**